[BUG] misc_expr_test FAILED on Dataproc

**Describe the bug**

Build ID: rapids_it-Dataproc/288

```
  integration_tests/src/main/python/misc_expr_test.py::test_mono_id FAILED [ 50%]
  integration_tests/src/main/python/misc_expr_test.py::test_part_id PASSED [100%]
  
  =================================== FAILURES ===================================
  _________________________________ test_mono_id _________________________________
  
      def test_mono_id():
  >       assert_gpu_and_cpu_are_equal_collect(
                  lambda spark : unary_op_df(spark, short_gen).select(
                      f.col('a'),
                      f.monotonically_increasing_id()))
  
  integration_tests/src/main/python/misc_expr_test.py:24: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  integration_tests/src/main/python/asserts.py:440: in assert_gpu_and_cpu_are_equal_collect
      _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
  integration_tests/src/main/python/asserts.py:432: in _assert_gpu_and_cpu_are_equal
      assert_equal(from_cpu, from_gpu)
  integration_tests/src/main/python/asserts.py:101: in assert_equal
      _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
  integration_tests/src/main/python/asserts.py:42: in _assert_equal
      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
  integration_tests/src/main/python/asserts.py:35: in _assert_equal
      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  cpu = 512, gpu = 8589934592
  float_check = <function get_float_check.<locals>.<lambda> at 0x7fc794309790>
  path = [512, 'monotonically_increasing_id()']
  
      def _assert_equal(cpu, gpu, float_check, path):
          t = type(cpu)
          if (t is Row):
              assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
              if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
                  assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)
                  for field in cpu.__fields__:
                      _assert_equal(cpu[field], gpu[field], float_check, path + [field])
              else:
                  for index in range(len(cpu)):
                      _assert_equal(cpu[index], gpu[index], float_check, path + [index])
          elif (t is list):
              assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
              for index in range(len(cpu)):
                  _assert_equal(cpu[index], gpu[index], float_check, path + [index])
          elif (t is tuple):
              assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
              for index in range(len(cpu)):
                  _assert_equal(cpu[index], gpu[index], float_check, path + [index])
          elif (t is pytypes.GeneratorType):
              index = 0
              # generator has no zip :( so we have to do this the hard way
              done = False
              while not done:
                  sub_cpu = None
                  sub_gpu = None
                  try:
                      sub_cpu = next(cpu)
                  except StopIteration:
                      done = True
      
                  try:
                      sub_gpu = next(gpu)
                  except StopIteration:
                      done = True
      
                  if done:
                      assert sub_cpu == sub_gpu and sub_cpu == None, "CPU and GPU generators have different lengths at {}".format(path)
                  else:
                      _assert_equal(sub_cpu, sub_gpu, float_check, path + [index])
      
                  index = index + 1
          elif (t is dict):
              # The order of key/values is not guaranteed in python dicts, nor are they guaranteed by Spark
              # so sort the items to do our best with ignoring the order of dicts
              cpu_items = list(cpu.items()).sort(key=_RowCmp)
              gpu_items = list(gpu.items()).sort(key=_RowCmp)
              _assert_equal(cpu_items, gpu_items, float_check, path + ["map"])
          elif (t is int):
  >           assert cpu == gpu, "GPU and CPU int values are different at {}".format(path)
  E           AssertionError: GPU and CPU int values are different at [512, 'monotonically_increasing_id()']
  
  integration_tests/src/main/python/asserts.py:77: AssertionError
  ----------------------------- Captured stdout call -----------------------------
```
**Environment details (please complete the following information)**
 - Environment location: Google Dataproc

--- a/integration_tests/src/main/python/misc_expr_test.py
+++ b/integration_tests/src/main/python/misc_expr_test.py
@@ -1,4 +1,4 @@
-# Copyright (c) 2020, NVIDIA CORPORATION.
+# Copyright (c) 2020-2021, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -22,12 +22,12 @@ import pyspark.sql.functions as f
 
 def test_mono_id():
     assert_gpu_and_cpu_are_equal_collect(
-            lambda spark : unary_op_df(spark, short_gen).select(
+            lambda spark : unary_op_df(spark, short_gen, num_slices=8).select(
                 f.col('a'),
                 f.monotonically_increasing_id()))
 
 def test_part_id():
     assert_gpu_and_cpu_are_equal_collect(
-            lambda spark : unary_op_df(spark, short_gen).select(
+            lambda spark : unary_op_df(spark, short_gen, num_slices=8).select(
                 f.col('a'),
                 f.spark_partition_id()))

